Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -957,18 +963,28 @@ bool MatrixWorkspace::isCommonBins() const {
   if (!m_isCommonBinsFlagSet) {
     m_isCommonBinsFlag = true;
 
+    const size_t numHist = getNumberHistograms();
+    bool oneNonEmpty = false;
+    if (numHist > 1) {
+      for (size_t i = 0; i < numHist; ++i) {
+        if (!y(i).empty()) {
+          oneNonEmpty = true;
+          break;
+        }
+      }
+    }
+
     // there being only one or zero histograms is accepted as not being an error
-    if (blocksize() || getNumberHistograms() > 1) {
+    if (oneNonEmpty || numHist > 1) {
       // otherwise will compare some of the data, to save time just check two
       // the first and the last
```

Comment:
I think this is now definitely not enough any more, is it? Maybe at least the length should be checked, and if it it all the same continue with the existing "test" via first and last?